### ISSUE TITLE
Reference to undeclared resource

### ISSUE THREAD

**Issue by luisarnauth** [Azure/terraform-azurerm-avm-res-web-site#42]
While trying to deploy a Linux Web App with IP restrictions, I get these errors:

```terraform
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Reference to undeclared resource
│ 
│   on .terraform/modules/web_appe/main.web_app.tf line 625, in resource "azurerm_linux_web_app" "this":
│  625:         ip_address                = scm_ip_restriction.value.ip_address
│ 
│ A managed resource "scm_ip_restriction" "value" has not been declared in module.web_appe.
╵
╷
│ Error: Reference to undeclared resource
│ 
│   on .terraform/modules/web_appe/main.web_app.tf line 626, in resource "azurerm_linux_web_app" "this":
│  626:         name                      = scm_ip_restriction.value.name
│ 
│ A managed resource "scm_ip_restriction" "value" has not been declared in module.web_appe.
╵
╷
│ Error: Reference to undeclared resource
│ 
│   on .terraform/modules/web_appe/main.web_app.tf line 627, in resource "azurerm_linux_web_app" "this":
│  627:         priority                  = scm_ip_restriction.value.priority
│ 
│ A managed resource "scm_ip_restriction" "value" has not been declared in module.web_appe.
╵
╷
│ Error: Reference to undeclared resource
│ 
│   on .terraform/modules/web_appe/main.web_app.tf line 628, in resource "azurerm_linux_web_app" "this":
│  628:         service_tag               = scm_ip_restriction.value.service_tag
│ 
│ A managed resource "scm_ip_restriction" "value" has not been declared in module.web_appe.
╵
╷
│ Error: Reference to undeclared resource
│ 
│   on .terraform/modules/web_appe/main.web_app.tf line 629, in resource "azurerm_linux_web_app" "this":
│  629:         virtual_network_subnet_id = scm_ip_restriction.value.virtual_network_subnet_id
│ 
│ A managed resource "scm_ip_restriction" "value" has not been declared in module.web_appe.
```

I believe the issue is on the "ip_restriction" dynamic block that wrongly references "scm_ip_restriction" instead of "ip_restriction". 

https://github.com/Azure/terraform-azurerm-avm-res-web-site/blob/503d05114ed7d31e527a9d8b2bf2d586cd7d1ea4/main.web_app.tf#L620-L642

The Windows Web App, and both Functions (Windows and Linux) have the same code.

**Comment by donovm4** [comment]
@luisarnauth - thank you, will look into this.

**Comment by donovm4** [comment]
bug fix addressed in v0.4.0 release
